python/tank/util/shotgun.py

```diff
@@ -411,11 +411,28 @@ def __get_app_store_connection_information():
     config_data["host"] = constants.SGTK_APP_STORE
     config_data["api_script"] = script_name
     config_data["api_key"] = script_key
-    config_data["http_proxy"] = client_site_sg.config.raw_http_proxy
+    config_data["http_proxy"] = __get_app_store_proxy_setting(client_site_sg)
 
     return config_data
 
 
+def __get_app_store_proxy_setting(connection):
+    """
+    Retrieve the app store proxy settings. If the key
+    app_store_http_proxy is not found in the shotgun.yml file, the proxy
+    settings from the client site connection will be used.
+
+    :param connection: A working connection to the site.
+
+    :returns: The http proxy connection string.
+    """
+    config_data = get_associated_sg_config_data()
+    if config_data and config_data.get("app_store_http_proxy"):
```

Comment:
I think using a constant here will make make things clearer. Like we do with `constants.SGTK_APP_STORE` above.